src/linux.rs

```diff
@@ -0,0 +1,95 @@
+// Copyright 2016 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under (1) the MaidSafe.net
+// Commercial License, version 1.0 or later, or (2) The General Public License
+// (GPL), version 3, depending on which licence you accepted on initial access
+// to the Software (the "Licences").
+//
+// By contributing code to the SAFE Network Software, or to this project
+// generally, you agree to be bound by the terms of the MaidSafe Contributor
+// Agreement, version 1.0.
+// This, along with the Licenses can be found in the root directory of this
+// project at LICENSE, COPYING and CONTRIBUTOR.
+//
+// Unless required by applicable law or agreed to in writing, the SAFE Network
+// Software distributed under the GPL Licence is distributed on an "AS IS"
+// BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+//
+// Please review the Licences for the specific language governing permissions
+// and limitations relating to use of the SAFE Network Software.
+
+
+use std::env;
+use std::ascii::AsciiExt;
+use std::fs::{create_dir_all, File};
+use std::io::Write;
+use std::process::Command;
+use std::path::PathBuf;
+
+use app::App;
+
+
+/// Open a given URI on Linux systems
+pub fn open(uri: String) -> bool {
+    Command::new("xdg-open")
+        .arg(uri)
+        .status()
+        .expect("failed to start xdg-open")
+        .success()
+}
+
+
+/// register the given App for the given schemes on Linux
+pub fn install(app: App, schemes: Vec<String>) -> bool {
+    if let Some(home) = env::home_dir() {
+        let ascii_name = format!("{}-{}",
+                                 app.vendor.as_str().to_ascii_lowercase(),
+                                 app.name.as_str().to_ascii_lowercase());
+
+        let mut desktop_target = PathBuf::new();
+        desktop_target.push(home);
+        desktop_target.push(".local");
+        desktop_target.push("share");
+        desktop_target.push("applications");
+
+        let apps_dir = desktop_target.clone();
+
+        if create_dir_all(apps_dir.clone()).is_ok() {
```

Comment:
Instead of so many levels of indentation as with `if/else` case as below, return proper error (`SysUriError` as stated in this review). Then things should just reduce to:
```
let abc = foo()?;
let def = bar(acb)?;
...
```
pattern. Do aim for that pattern to multilevel indents. Also that will return proper error instead of just false, when things fail giving better diagnostics and better help decide recovery mechanisms if any.